app/components/download-graph.js

```diff
@@ -67,81 +75,109 @@ export default class DownloadGraph extends Component {
   }
 
   get data() {
-    let downloads = this.args.data;
-    if (!downloads) {
-      return { datasets: [] };
-    }
+    return toChartData(this.args.data, this.args.versions);
+  }
+}
 
-    let extra = downloads.content?.meta?.extra_downloads ?? [];
+export function toChartData(data) {
+  if (!data) {
+    return { datasets: [] };
+  }
 
-    let dates = {};
-    let versions = new Set([]);
+  let extra = data.content?.meta?.extra_downloads ?? [];
 
-    let now = new Date();
-    for (let i = 0; i < 90; i++) {
-      let date = subDays(now, i);
-      dates[date.toISOString().slice(0, 10)] = { date, cnt: {} };
-    }
+  let dates = {};
+  let versions = new Map();
+  let crate = null;
 
-    downloads.forEach(d => {
-      let version = d.version;
-      if (!version) return;
+  let now = new Date();
+  for (let i = 0; i < 90; i++) {
+    let date = subDays(now, i);
+    dates[date.toISOString().slice(0, 10)] = { date, cnt: {} };
+  }
 
-      let version_num = version.num;
+  data.forEach(d => {
+    let version = d.version;
+    if (!version) return;
 
-      versions.add(version_num);
+    let version_num = version.num;
 
-      let key = d.date;
-      if (dates[key]) {
-        let prev = dates[key].cnt[version_num] || 0;
-        dates[key].cnt[version_num] = prev + d.downloads;
-      }
-    });
+    versions.set(version_num, version);
```

Comment:
Is there a reason this uses underscored variables over camelCase like most of the other code here?

Reply:
only legacy reasons 😉